### PR TITLE
Support optional query params

### PR DESCRIPTION
**Pull request by tronroy**
Previously ruby client did not allow users to pass in optional params like `initialInvisibilityTime` for publishing and getting messages, or `errorIfExists` for creating queues.

### CODE REVIEW

File: lib/cassieq/client/messages.rb [hunk 1]
```diff
@@ -3,16 +3,18 @@
 module Cassieq
   class Client
     module Messages
-      def publish_message(queue_name, message)
-        request(:post, "queues/#{queue_name}/messages", message)
+      def publish_message(queue_name, message, initial_invis_time = nil)
+        query = { initialInvisibilityTime: initial_invis_time } unless initial_invis_time.nil?
```
Comment (devshorts) [inline lib/cassieq/client/messages.rb]: We want this to be initialize invisiblity seconds, the `time` suffix is deprecated

File: spec/cassettes/statistics/statistics.yml [hunk 2]
```diff
@@ -10,9 +10,9 @@ http_interactions:
       User-Agent:
       - Faraday v0.9.2
       X-Cassieq-Request-Time:
-      - '2016-04-12T02:03:11Z'
+      - '2016-04-18T19:22:53Z'
       Authorization:
-      - Signed smxpy9my1EIInq6nKc6Fcig5Ws2kjJ5-1jq0nMNVBBg
+      - Signed GdzpbmzPdzpXyb-Pdy45IPi7_dYB9rBiKf1jY50JWLw
```
Comment (devshorts) [inline spec/cassettes/statistics/statistics.yml]: Out of curiosoity what are these files? Are they auto generated?

File: lib/cassieq/client/messages.rb (hunk 1 above)
Comment (tronroy) [inline lib/cassieq/client/messages.rb]: Oops, got mixed up.

**Comment by devshorts** [comment]
Overall looks good, but the deprecated parameter needs to change :). 

I'd like to actually see your tests pull in the cassieq docker container, run it local, and use your api to interact with it.  You can update your travis file like this: https://github.com/paradoxical-io/rabbitmq/blob/master/.travis.yml#L16,L17 to support docker when you build in travis.  There are probably ruby docker clients out there that will let you pull a container, configure a port to open, etc. You can use the cassieq dev mode to spin up the api and run it. 

This way you don't need to hardcode keys, etc in your specs. Each test run you can just create new accounts/queues with the client.

File: lib/cassieq/client/messages.rb (hunk 1 above)
Comment (devshorts) [inline lib/cassieq/client/messages.rb]: No worries. In general its always good to include the unit type if the data type itself isn't expressive. This was a bad decision on our part to not include the time unit in the name originally

File: spec/cassettes/statistics/statistics.yml (hunk 2 above)
Comment (tronroy) [inline spec/cassettes/statistics/statistics.yml]: Most of those files are related to vcr https://github.com/vcr/vcr which I use to record text fixtures.

Comment (devshorts) [inline spec/cassettes/statistics/statistics.yml]: Should they be git ignored?

Comment (tronroy) [inline spec/cassettes/statistics/statistics.yml]: They are needed to run most of the test, atm.

Comment (devshorts) [inline spec/cassettes/statistics/statistics.yml]: sounds good, just checkin :)

**Comment by tronroy** [comment]
Yeah, that sounds like a better approach to testing than how I currently do it, i.e. recording http interactions against a local cassieq container using the VCR gem.

File: spec/cassettes/statistics/statistics.yml (hunk 2 above)
Comment (tronroy) [inline spec/cassettes/statistics/statistics.yml]: I agree it's not ideal. I'll look into implementing the testing procedure you describe below. Thanks :)

Comment (devshorts) [inline spec/cassettes/statistics/statistics.yml]: No sweat, its all about iteration. I'm just glad you have solid tests :) Thats more than most!  Thanks for the hard work tom!

**Comment by tronroy** [comment]
Changed the parameter, good to merge?

**Comment by devshorts** [comment]
LGTM

**Pull request merged**